### DISCUSSION TITLE
Support JIT ( Just In Time )

### DISCUSSION THREAD

**Discussion by suiramdev** [tailwindlabs/headlessui#1371] [Ideas]
HeadlessUI doesn't seem to support the new Just In Time feature from TailwindCSS.
On my project, as an example, I need the Just In Time feature to delay my transitions based on a value from 1 to infinity. However, this cannot be done without JIT.
![image](https://user-images.githubusercontent.com/71449142/165633080-24372cf3-e64a-4671-8470-f5b56833472a.png)

Please, add support to it :).

**Comment by RobinMalfait** [comment]
Hey! Thank you for your question!
Much appreciated! 🙏

This is not what "Just In Time" means in Tailwinds context. Tailwind is still a build tool, it will generate the necessary utilities when you use them in your code. But we can't know what `index` is referring to at runtime because Tailwind was already compiled.

In other words, don't do this `delay-[${index}s]`. More information can be found here:

- https://tailwindcss.com/docs/content-configuration#dynamic-class-names
- https://tailwindcss.com/docs/content-configuration#class-detection-in-depth
